docs/DeveloperGuide.md

```diff
@@ -1,34 +1,600 @@
+    delete patient S7777777Z 
+    
+will delete the patient with NRIC `S7777777Z`, if found. 
+
+**Implementation** 
+
+![Delete Patient Sequence Diagram](images/DeletePatientSequenceDiagram.jpg)
+
+The following steps explains the sequence of events: 
+
+1. The user enters `delete patient S7777777Z`. 
+
+2. `HappyPills` calls `Parser#parse()` which then calls `PatientParser#parse()`.
+
+3. `PatientParser#parse()` will then create an object `DeletePatientCommand`. 
+
+4. `HappyPills` then calls `DeletePatientCommand.execute()` to execute the command. 
+
+5. In `DeletePatientCommand.execute()`, the respective patient is deleted and the display message is returned. 
+
+#### 4.1.4. List Patients
+
+The user is able to get a list of all the patients currently in the program.
+The command: 
+    
+    list patient
+    
+will list all the appointments in the `PatientMap`. 
+
+**Implementation** 
+
+The following steps explains the sequence of events: 
+
+1. The user enters `list patient`.
+
+2. `HappyPills` calls `Parser#parse()` which then calls `PatientParser#parse()`.
+
+3. `APatientParser#parse()` will create an object `ListPatientCommand`.
+
+4. `HappyPills` then calls `ListPatientCommand.execute()` to execute the method. 
+
+5. In `ListPatientCommand.execute()`, it calls `TextUi.getPatientList()` which is then returned as the display message. 
+
+#### 4.1.5. Retrieve Patient Details 
+
+The user can retrieve a patient's details. The command: 
+
+    get patient S7777777Z
+    
+will retrieve the details of the patient with NRIC `S7777777Z`. 
+
+**Implementation** 
+
+![Get Patient Sequence Diagram](images/GetPatientSequenceDiagram.jpg)
+
+The following steps explains the sequence of events: 
+
+1. The user enters `get patient S7777777Z` .
+
+2. `HappyPills` calls `Parser#parse()` which then calls `PatientParser#parse()`. 
+
+3. `PatientParser#parse()` will create an object `GetPatientCommand`.
+
+5. `HappyPills` then calls `GetPatientCommand.execute()` to execute the method. 
+
+6. `GetPatientCommand.execute()` calls `TextUi.getPatientSuccessMessage()` if a patient of NRIC `S7777777Z` can be found 
+or `TextUi.patientNotExists()` if no patient with the given NRIC exists in the program. 
+
+7. The respective `TextUi` methods will then display the message to the user accordingly.
+
+### 4.2 Patient Medical Records Features 
+
+
+### 4.3. Appointment Scheduling Feature 
+
+#### 4.3.1. Add Appointment
+
+The user is able to add appointments into the program to manage the appointment schedule. 
+The command: 
+
+    add appt /ic S7777777Z /d 01/04/2020 /t 12:00:00 /r Checkup 
+    
+will add an appointment with `NRIC` as S7777777Z with the following attributes: 
+* appointment date: `01/04/2020`
+* appointment time: `12:00:00`
+* reason for appointment: `Checkup`
+
+An `appointmentId` will also be given when an appointment is successfully added. 
+
+**Implementation** 
+
+![Add Appointment Sequence Diagram](images/AddAppointmentSequenceDiagram.jpg)
```

Comment:
![image](https://user-images.githubusercontent.com/51941453/78000072-446bdf00-7366-11ea-9135-5410a12104e0.png)

Same comments as above diagrams since they follow a similar format